Compiler/Util/BaseAvlSet.mo

```diff
@@ -134,6 +156,7 @@ algorithm
   key := match inTree
     case EMPTY() algorithm return; then fail();
     case NODE() then inTree.key;
+    case LEAF() then inTree.key;
```

Comment:
That would be a compile-time error (because we do not know if it is NODE or LEAF